Draft-Accepted/RFCNNNN-Get-WhatsNew.md

```diff
@@ -0,0 +1,231 @@
+- Supports a message-of-the-day (motd) option
+
+Non-goals/Alternatives
+
+- Data endpoint (Microservice) to synchronously deliver or update the data.
+- Changelog information
+
+Future feature considerations
+
+- Render markdown output in the console with ANSI styling (enabled by switch parameter)
+- Use ANSI strings to make hyperlinks clickable for terminals that support it
+- Add `Get-Changelog` cmdlet to display Changelog information
+
+## Solution
+
+Ship a **Microsoft.PowerShell.WhatsNew** module on the PowerShell Gallery that contains the cmdlet
+and would:
+
+- Be discoverable using normal PowerShell conventions
+- Be available down-level to Windows PowerShell 5.1
+
+## Specification
+
+This specification is based on the cmdlet [Get-WhatsNew](https://github.com/PowerShell/WhatsNew)
+
+- Module Name: Microsoft.PowerShell.WhatsNew
+- Cmdlet name: Get-WhatsNew
+- Alias: none
+
+### Syntax and parameter sets
+
+- ByVersion (Default)
+
+```
+Get-WhatsNew [[-Version] <string>] [-Daily] [-Online] [<CommonParameters>]
+```
+
+- CompareVersion
+
+```
+Get-WhatsNew [[-Version] <string>] -CompareVersion <double> [<CommonParameters>]
+```
+
+- AllVersions
+
+```
+Get-WhatsNew -All [<CommonParameters>]
+```
+
+The **Version** parameter has the following validation attribute: `[ValidateSet('5.1','7.0','7.1','7.2','7.3')]`
+
+### Data model
+
+To minimize content maintenance and allow for disconnected scenarios, the data presented by the
+cmdlet is stored with the module as version specific markdown files.
+
+- Uses the same markdown files as the [webview](https://docs.microsoft.com/powershell/scripting/whats-new/what-s-new-in-powershell-72)
+- Version files are updated in the module at the time of a new release
+- This allows for disconnected scenarios
+- Minimizes the maintenance of the content
+- Update support
+  - Updates to the data require publishing a new version of the module to the Gallery
+  - To get the latest data, the user runs `Update-Module`
+  - Eliminates the need for a Microservice to serve the data or updates
+
+### Output
+
+Output from the cmdlet is in the form of markdown strings that are displayed to the terminal.
+
+### Demo.txt - Features
+
+- To list the new features released in the current version:
+
+  ```powershell
+  Get-WhatsNew
+  ```
+
+  Example output:
+
+  ```output
+  (truncated for clarity)
+  ## Installation updates
+
+  Check the installation instructions for your preferred operating system:
+
+  - [Windows][Windows]
+  - [macOS][macOS]
+  - [Linux][Linux]
+
+  Additionally, PowerShell 7.2 supports ARM64 versions of Windows and macOS and ARM32 and ARM64
+  versions of Debian and Ubuntu.
+
+  For up-to-date information about supported operating systems and support lifecycle, see the
+  [PowerShell Support Lifecycle][lifecycle].
+  (truncated for clarity)
+  ```
+
+- To receive a **Message Of The Day** (motd) displaying a single new feature
+
+  ```powershell
+  Get-WhatsNew -Daily
```

Comment:
need to understand how daily doesn't repeat and presumably is random.  probably needs some state stored somewhere so it doesn't repeat the same motd

Reply:
I am using the system time as a random seed. This should keep it from repeating as best as we can. We can evaluate this more later. It may not be worth the effort.